prow/apis/prowjobs/v1/types.go

```diff
@@ -404,6 +404,9 @@ type DecorationConfig struct {
 	// after sending SIGINT to send SIGKILL when aborting
 	// a job. Only applicable if decorating the PodSpec.
 	GracePeriod *Duration `json:"grace_period,omitempty"`
+	// UseDefaultEntrypoint indicates whether entrypoint should try to discover
+	// the default entrypoint from the image
+	UseDefaultEntrypoint *bool `json:"use_default_entrypoint,omitempty"`
```

Comment:
Isn't this implicit by `command` being unset?

Reply:
right now prow is generous enough to combine `command` and `args`, then pick the `[0]` as executable, so it's totally fine for a job to set executable as the first arg under `args`. I have done a quick search in this repo and there are 140+ jobs going down this route, so relying on `command` being unset would be a breaking change. (it's trivial to change though)

on the other hand, the initial idea is to start this feature as `opt-in` mode, so that it's not surprising for scenarios that this feature doesn't work well. If it proved to work well we can migrate to `opt-out` mode